### PR TITLE
_includes: openring: enable openring.

### PR DESCRIPTION
**Pull request by aquilamacedo**
This commit creates a openring page based on the build time, using jekyll hooks.

Signed-off-by: Aquila Macedo <aquilamacedo@riseup.net>

### CODE REVIEW

File: _plugins/hook_list.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# https://jekyllrb.com/docs/plugins/hooks/
+Jekyll::Hooks.register :site, :post_write do
+    # code to call after Jekyll:
+    system('cd openring && ./openring.sh')
```
Comment (rodrigosiqueira) [inline _plugins/hook_list.rb]: How about this draft:

```ruby
require 'mkmf'

OPENRING_LIST="openring/feed_list.txt"
OPENRING_IN="openring/openring_template_in.html"
OPENRING_OUT="_includes/openring_template_out.html"

def composing_openring_command(openring_path)
  openring_custom_parameters = ''

  File.foreach(OPENRING_LIST) do |line|
    openring_custom_parameters += " -s #{line.chomp}"
  end

  return "#{openring_path} #{openring_custom_parameters} < #{OPENRING_IN} > #{OPENRING_OUT}"
end

Jekyll::Hooks.register :site, :post_write do
    openring_path = find_executable('openring')
    openring_cmd = composing_openring_command(openring_path)

    system(openring_cmd)
end
```

With this code, you can delete the openring.sh file. But you need to improve the above code by:
* Add checks before accessing a file
* Check the OPENRING_IN file timestamp. If it was generated 15 min ago, there is no need to run this script. 
* Add documentation

File: openring/openring.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (rodrigosiqueira) [inline openring/openring.sh]: We can delete this script, see my comment in the plugin :)

File: _includes/openring_template_out.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
+
```
Comment (rodrigosiqueira) [inline _includes/openring_template_out.html]: Since this file is dynamically generated, there is no reason to commit it. Just add it to the `.gitigore` file.

File: openring/openring_template_in.html [hunk 4]
```diff
@@ -36,11 +41,15 @@ <h4 class="title">
   padding: 0.5rem;
   background: #eee;
   min-width: 10rem;
+  padding: 0.5rem;
+  margin: 0 0.5rem;
```
Comment (rodrigosiqueira) [inline openring/openring_template_in.html:45]: Let's make the background transparent and a thin board with our default green color. Something like:
```
border-width: 1px;
border-style: solid;
border-color:
```
In the responsive mode, make sure that you add a proper space.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: _includes/openring_out.html [hunk 5]
```diff
@@ -1,74 +0,0 @@
-<section class="webring">
```
Comment (rodrigosiqueira) [inline _includes/openring_out.html:1]: Why is this file here? We don't have openring_out.html in the master, only openring_template_out.html. I think this is something local to your branch.

File: _includes/openring_template_out.html (hunk 3 above)
Comment (rodrigosiqueira) [reply _includes/openring_template_out.html]: Did you address my previous comment? Should this file be part of the .gitignore?

File: _layouts/post.html [hunk 6]
```diff
@@ -28,9 +28,9 @@ <h1> {{ page.title }} </h1>
 
   <div align="center">
     {% include share.html %}
+    {% include openring_template_out.html %}
```
Comment (rodrigosiqueira) [inline _layouts/post.html:31]: I think we need an `if` statement here that checks if this file exists before including it.

File: _plugins/opering_hook.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  return "#{openring_path} #{openring_custom_parameters} < #{OPENRING_IN} > #{OPENRING_OUT}"
+end
+
+Jekyll::Hooks.register :site, :post_write do
+  if file_was_created_more_than_x_min?
+    puts("-" * 80)
+
+    puts("[!] keep calm, we're running openring...")
+    openring_path = find_executable('openring')
```
Comment (rodrigosiqueira) [inline _plugins/opering_hook.rb]: What if `openring` does not exist? You need to take action in this case, and I think this should be the first verification, even before this if.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi @aquilamacedo ,
I have already reviewed your new change; see my comments. Also, update the commit message to make it more descriptive.

File: _includes/openring_out.html (hunk 5 above)
Comment (aquilamacedo) [reply _includes/openring_out.html:1]: Actually we have this file in _includes/openring_out.html in the master, I removed it because we probably won't use it again.

image

File: _layouts/post.html (hunk 6 above)
Comment (aquilamacedo) [reply _layouts/post.html:31]: I just made this change and I created this instruction inside the openring_hook.rb file.

**Comment by rodrigosiqueira** [comment]
Merged! Thanks a lot!
